packages/attestation-service/src/requestHandlers/attestation.ts

```diff
@@ -68,18 +70,28 @@ class AttestationRequestHandler {
       throw new Error(ATTESTATION_ALREADY_SENT_ERROR)
     }
 
-    if (!isDevMode()) {
-      const attestations = await kit.contracts.getAttestations()
-      const state = await attestations.getAttestationState(this.key.identifier, account, issuer)
+    if (isDevMode()) {
+      return
+    }
 
-      if (state.attestationState !== AttestationState.Incomplete) {
-        Counters.attestationRequestsWOIncompleteAttestation.inc()
-        throw new Error(NO_INCOMPLETE_ATTESTATION_FOUND_ERROR)
+    // Check the on-chain status of the attestation. If it's marked Complete, don't do it.
+    // If it's missing, the full node could be behind by a block or two. Try a few times before erroring.
```

Comment:
Would it be better to check the age of the latest block here?

Reply:
Have we been seeing issues for validators that the attestation is missing because the full node is behind?  We already wait 5 blocks worth between when the client requests the issuers and it selects them. Is waiting 2 more blocks worth likely to help those cases or are they typically very behind?

Reply:
We have seen it last month. After calling `selectIssuers` we don't wait, and it's the result of that that needs to be visible to the Attestation Service

Reply:
sounds like we need to dig into why the client is not waiting